File: src/main/java/uk/laxd/dndSimulator/feature/StaticFeature.java
package uk.laxd.dndSimulator.feature;

/**
 *  A {@code Feature} that is always active, and applies whenever a condition is met, i.e. Brutal Critical
 */
public class StaticFeature extends Feature {
    public StaticFeature(String name) {
        super(name);
    }

    @Override
    public boolean isActiveSkill() {
        return false;
    }

    @Override
    public boolean isActive() {
        return true;
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/feature/StaticFeature.kt
package uk.laxd.dndSimulator.feature

/**
 * A `Feature` that is always active, and applies whenever a condition is met, i.e. Brutal Critical
 */
abstract class StaticFeature(name: String) : Feature(name) {
    override val isActiveSkill: Boolean = false
    override var isActive: Boolean = true
}